server/header_context.go

```diff
@@ -104,15 +106,19 @@ func apigeeDynamicDataHeaders(org, env, api string, apiSpec *config.APISpec, fau
 	}
 
 	// Include fault related headers.
-	if fault {
+	if adapterFault != nil && adapterFault.RpcCode == rpc.INTERNAL {
 		headers = append(headers, createHeaderValueOption(headerFaultSource, "ARC", false))
 		headers = append(headers, createHeaderValueOption(headerFaultFlag, "true", false))
 		if apiSpec != nil {
 			headers = append(headers, createHeaderValueOption(headerFaultRevision, apiSpec.RevisionID, false))
 		}
 
-		// A placeholder fault code value.
-		headers = append(headers, createHeaderValueOption(headerFaultCode, "fault", false))
+		if adapterFault.FaultCode == "" {
```

Comment:
Like here, you could define an accessor on `AdapterFault` that returns `"fault"` (or "defaultFaultCode", or whatever you like) if there is no `FaultCode` defined, and the value of `FaultCode` otherwise

Reply:
Thanks for the suggestion, I am not sure if I'd want adapterFault to return `fault` as the default value. The only reason I am keeping this here is to keep the existing logic unchanged.
Skipping this for now.